### PR TITLE
fix: add OpenCode slash command discovery and execution support

### LINKED ISSUE #552: Opencode slash commands configured by user are not recognized from Maestro chat
### Summary
When using Maestro with **opencode**, slash commands configured for opencode are not recognized when typed in chat. The input appears to be treated as regular user text rather than an agent-native slash command.

### Repro (Desktop)
1. Create/use an opencode session.
2. Configure an opencode native slash command in the tool (or environment, plugin/config depending on how it is registered).
3. In Maestro chat, type `/your-command ...`.
4. Observe: the command does not execute as a slash command in opencode and is handled as normal message flow.

### Suspected root cause
This appears to be by design in current code paths:

1. Opencode is currently marked as not supporting native slash commands:
- `src/main/agents/capabilities.ts` (`opencode` block): `supportsSlashCommands: false`.

2. Slash-command discovery IPC is hard-gated to Claude Code only:
- `src/main/ipc/handlers/agents.ts` (`agents:discoverSlashCommands`) returns `null` when `agentId !== 'claude-code'`.
- This means `activeSession.agentCommands` can never be populated for opencode via this route.

3. UI only exposes discovered agent commands when the capability flag is true:
- `src/renderer/App.tsx` around `agentCommands = hasActiveSessionCapability('supportsSlashCommands') ? ... : []`.

4. Input handling for slashes only treats:
- built-in Maestro `/history`, `/wizard`, `/skills` (Claude-only),
- plus `customAICommands`.
- unmatched slash input falls through to normal message flow, which then gets wrapped as user prompt text.
  - In `src/renderer/hooks/input/useInputProcessing.ts`, non-command path eventually builds `effectivePrompt` and appends `maestroSystemPrompt` for new sessions.

### Why this breaks opencode slash workflows
If opencode depends on raw slash-prefixed commands from chat, it currently receives a normal prompt text path (and may be prefixed in new sessions), so native command dispatch is never triggered.

### Probable fixes (ranked)
1. **Add opencode slash-command discovery path**
- Implement discovery for opencode if/where it exposes configured commands (likely not via Claude init event).
- Store result on session `agentCommands`.
- Gate execution path by capability only when this is true for opencode.

2. **Add passthrough execution path for configured opencode slash commands**
- If command is recognized as agent-native, send raw command input through the same message queue as a command item instead of converting to a normal text prompt.
- Avoid wrapping with `maestroSystemPrompt` for this special case.

3. **Clarify expected behavior in docs**
- If opencode does not support native slash commands in Maestro’s batch mode, make that explicit in `docs/provider-notes.md` and `docs/slash-commands.md` as a known limitation.

### Notes
- There is existing parser support for opencode output (`src/main/parsers/opencode-output-parser.ts`), but command discovery/execution is currently blocked by capability/design guards.

### Suggested acceptance
- With configured opencode slash command `/foo`, command appears in autocomplete (if discoverable) and executes without being wrapped by Maestro system prompt.
- Commands configured but unsupported should fail with clear feedback, not be treated as normal text silently.

### PR DESCRIPTION
**Pull request by chr1syy**
## Summary

- Enables OpenCode slash command discovery by reading commands from disk (`.opencode/commands/*.md`, `~/.config/opencode/commands/*.md`, and `opencode.json` config)
- Sets `supportsSlashCommands: true` for OpenCode agent so commands appear in autocomplete
- Makes `getSlashCommandDescription()` agent-aware to provide correct descriptions per agent

Closes #552

## Changes

| File | Change |
|------|--------|
| `src/main/agents/capabilities.ts` | Flip `supportsSlashCommands` to `true` for opencode |
| `src/main/ipc/handlers/agents.ts` | Add `discoverOpenCodeSlashCommands()` — disk-based discovery from 3 sources (built-in, project-local, global + config) |
| `src/renderer/constants/app.ts` | Add `OPENCODE_BUILTIN_COMMANDS` descriptions, make `getSlashCommandDescription()` agent-aware |
| `src/renderer/hooks/agent/useAgentListeners.ts` | Pass `toolType` to `getSlashCommandDescription()` |
| `src/renderer/hooks/wizard/useWizardHandlers.ts` | Pass `toolType` to `getSlashCommandDescription()` |

## How it works

Unlike Claude Code (which discovers commands via `system/init` JSON event), OpenCode stores commands on disk:
1. **Built-in**: `init`, `review`, `undo`, `redo`, `share`, `help`, `models`
2. **Project-local**: `.opencode/commands/<name>.md` files
3. **Global**: `~/.config/opencode/commands/<name>.md` files
4. **Config**: `opencode.json` → `command` property

The discovery function reads all three sources and returns a unified command list, matching Claude Code's end-to-end pipeline (discover → autocomplete → execute).

## Test plan

- [ ] Create an OpenCode session in Maestro and verify slash commands appear in autocomplete
- [ ] Add a custom `.opencode/commands/test.md` file and verify it's discovered
- [ ] Verify Claude Code slash commands still work unchanged
- [ ] Verify Codex sessions (no slash commands) are unaffected

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * OpenCode now exposes built-in slash commands: init, review, undo, redo, share, help, models.
  * Automatic discovery of custom OpenCode slash commands from project/global command files and opencode.json.

* **Improvements**
  * Command descriptions are resolved per agent/tool for accurate help and UI display.
  * Slash-command discovery runs for both Claude Code and OpenCode where applicable.

* **Tests**
  * Expanded tests covering OpenCode discovery, parsing, error paths, and edge cases.

### CODE REVIEW

File: src/main/ipc/handlers/agents.ts [hunk 1]
```diff
@@ -27,6 +29,97 @@ const handlerOpts = (
...
+			}
+		}
+	} catch (error) {
+		logger.debug(`Failed to read project OpenCode commands from ${projectCommandsDir}`, LOG_CONTEXT, {
+			error: String(error),
+		});
+	}
+
+	// Global custom commands: ~/.config/opencode/commands/*.md
+	const globalCommandsDir = path.join(os.homedir(), '.config', 'opencode', 'commands');
```
[the remark this answers is not in the document]
Comment (chr1syy) [reply src/main/ipc/handlers/agents.ts]: These paths mirror where OpenCode itself stores its config and commands. If OpenCode respects XDG_CONFIG_HOME, we should too — but we need to match their behavior, not independently apply the spec. I'll check what OpenCode does and align accordingly.

**Comment by chr1syy** [comment]
@coderabbitai review
@greptileai re-review

**Comment by chr1syy** [comment]
@coderabbitai re-review please
@greptileai re-review please

**Comment by chr1syy** [comment]
OpenCode command still not being picked up, converting this to draft to get further work done.

**Comment by chr1syy** [comment]
The PR code will now pickup custom slash commands from OpenCode from their .md files.
Worked on manual tests.

**Comment by chr1syy** [comment]
@coderabbitai re-review please
